### PR TITLE
Promote deprecation levels for release 1.6.0

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: kotlinx-coroutines-core/common/src/flow/operators/Errors.kt [hunk 1]
```diff
@@ -60,35 +60,6 @@ public fun <T> Flow<T>.catch(action: suspend FlowCollector<T>.(cause: Throwable)
...
- * Cancellation exceptions that were caused by the direct [cancel] call are not handled by this operator.
- *
- * @suppress **Deprecated**: Use `catch { e -> if (predicate(e)) emitAll(fallback) else throw e }`
- */
-@Deprecated(
-    level = DeprecationLevel.ERROR,
-    message = "Use catch { e -> if (predicate(e)) emitAll(fallback) else throw e }",
-    replaceWith = ReplaceWith("catch { e -> if (predicate(e)) emitAll(fallback) else throw e }")
-)
-public fun <T> Flow<T>.onErrorCollect(
```
Comment (qwwdfsad) [inline kotlinx-coroutines-core/common/src/flow/operators/Errors.kt:84]: This one was `@FlowPreview`

**Comment by qwwdfsad** [comment]
`BroadcastChannel<T>.asFlow()` is left as warning deliberately for now

File: kotlinx-coroutines-core/common/src/flow/operators/Errors.kt [hunk 2]
```diff
@@ -60,35 +60,6 @@ public fun <T> Flow<T>.catch(action: suspend FlowCollector<T>.(cause: Throwable)
...
 
-/**
- * @suppress **Deprecated**: Use `(Throwable) -> Boolean` functional type
- */
-@Deprecated(
-    level = DeprecationLevel.ERROR,
-    message = "Use (Throwable) -> Boolean functional type",
-    replaceWith = ReplaceWith("(Throwable) -> Boolean")
-)
-public typealias ExceptionPredicate = (Throwable) -> Boolean
```
Comment (dkhalanskyjb) [inline kotlinx-coroutines-core/common/src/flow/operators/Errors.kt:71]: Do I understand correctly that type aliases don't leave a trace in target binaries, so it doesn't make sense to make them hidden, as they would simply become useless?

**Review by dkhalanskyjb — APPROVED** [review]

**Comment by qwwdfsad** [comment]
>Do I understand correctly that type aliases don't leave a trace in target binaries, so it doesn't make sense to make them hidden, as they would simply become useless?

Right. Also, it used to be `@FlowPreview`

**Pull request merged**